core/src/main/scala/cats/TraverseFilter.scala

```diff
@@ -203,4 +204,15 @@ object TraverseFilter {
   @deprecated("Use cats.syntax object imports", "2.2.0")
   object nonInheritedOps extends ToTraverseFilterOps
 
+  private[cats] def traverseFilterDirectly[G[_], A, B](
+    fa: IterableOnce[A]
+  )(f: A => G[Option[B]])(implicit G: StackSafeMonad[G]): G[Vector[B]] = {
+    fa.iterator.foldLeft(G.pure(Vector.empty[B])) { case (bldrG, a) =>
+      G.map2(bldrG, f(a)) {
+        case (acc, Some(b)) => acc :+ b
```

Comment:
have you benchmarked `:+` on `Vector` vs `Chain`? I think Chain *may* be faster since it doesn't have to copy each internal block of items.

Reply:
Yeah that's next on my [todo list](https://github.com/typelevel/cats/pull/4498#issuecomment-1810120330)